recipes/_hadoop_hdfs_ha_checkconfig.rb

```diff
@@ -27,6 +27,9 @@
 # We have dfs.nameservices, and now need to check them
 dfs_nameservices.each do |ns|
   # Start namenode checks
+  if node['hadoop']['hdfs_site']['dfs.ha.namenodes'] 
+    Chef::Application.fatal!("NameNode HA requires node['hadoop']['hdfs_site']['dfs.ha.namenodes.#{ns}']") unless node['hadoop']['hdfs_site'].key?("dfs\.ha\.namenodes\.#{ns}")
+  end
```

Comment:
That was really confusing to read. Anyway, there's no such property `dfs.ha.namenodes` in `hdfs-site.xml` which means your `if` statement should never be matched. It makes your entire code block pointless.

https://hadoop.apache.org/docs/r2.4.1/hadoop-project-dist/hadoop-hdfs/hdfs-default.xml

It sounds like you're attempting to setup HA without actually configuring all of the required options, but I really cannot tell.